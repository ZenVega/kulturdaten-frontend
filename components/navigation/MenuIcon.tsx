import styled from '@emotion/styled';
import React from 'react';

export enum MenuIconName {
  start = 'start',
  organizer = 'organizer',
  offer = 'offer',
  location = 'location',
  user = 'user',
}

const StartSvg: React.FC = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M5.25 9.75L12 4.5L18.75 9.75V18C18.75 18.3978 18.592 18.7794 18.3107 19.0607C18.0294 19.342 17.6478 19.5 17.25 19.5H6.75C6.35218 19.5 5.97064 19.342 5.68934 19.0607C5.40804 18.7794 5.25 18.3978 5.25 18V9.75Z"
      stroke="#111111"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M9.75 19.5V12H14.25V19.5"
      stroke="#111111"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);

const LocationSvg: React.FC = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <circle cx="12" cy="12" r="9.5" fill="#46948C" stroke="url(#paint0_linear)" />
    <defs>
      <linearGradient
        id="paint0_linear"
        x1="4.5"
        y1="5"
        x2="19"
        y2="19.5"
        gradientUnits="userSpaceOnUse"
      >
        <stop stopColor="#6FC1B9" />
        <stop offset="1" stopColor="#1B645C" />
      </linearGradient>
    </defs>
  </svg>
);

const OfferSvg: React.FC = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <rect
      x="1.58978"
      y="5.98959"
      width="17"
      height="17"
      rx="2.5"
      transform="rotate(-15 1.58978 5.98959)"
      fill="#273B8A"
      stroke="url(#paint1_linear)"
    />
    <defs>
      <linearGradient
        id="paint1_linear"
        x1="2.20215"
        y1="6.34314"
        x2="18.2021"
        y2="22.3431"
        gradientUnits="userSpaceOnUse"
      >
        <stop stopColor="#6984EF" />
        <stop offset="1" stopColor="#000B37" />
      </linearGradient>
    </defs>
  </svg>
);

const OrganizerSvg: React.FC = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M17.1823 4.32248C16.6842 2.46356 14.3606 1.84094 12.9997 3.20177L3.20178 12.9997C1.84095 14.3606 2.46357 16.6842 4.3225 17.1823L17.7068 20.7686C19.5657 21.2667 21.2667 19.5657 20.7686 17.7067L17.1823 4.32248Z"
      fill="#E60032"
      stroke="url(#paint2_linear)"
    />
    <defs>
      <linearGradient
        id="paint2_linear"
        x1="8.19702"
        y1="8.6969"
        x2="19.697"
        y2="20.1969"
        gradientUnits="userSpaceOnUse"
      >
        <stop stopColor="#E58EA1" />
        <stop offset="1" stopColor="#9F0B2B" />
      </linearGradient>
    </defs>
  </svg>
);

const UserSvg: React.FC = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M18 18.75V17.25C18 16.4544 17.6839 15.6913 17.1213 15.1287C16.5587 14.5661 15.7956 14.25 15 14.25H9C8.20435 14.25 7.44129 14.5661 6.87868 15.1287C6.31607 15.6913 6 16.4544 6 17.25V18.75"
      stroke="#111111"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M12 11.25C13.6569 11.25 15 9.90685 15 8.25C15 6.59315 13.6569 5.25 12 5.25C10.3431 5.25 9 6.59315 9 8.25C9 9.90685 10.3431 11.25 12 11.25Z"
      stroke="#111111"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);

const StyledMenuicon = styled.div`
  svg {
    display: block;
  }
`;

interface MenuIconProps {
  type: MenuIconName;
}

const icons: { [key in MenuIconName]: React.FC } = {
  start: StartSvg,
  organizer: OrganizerSvg,
  location: LocationSvg,
  offer: OfferSvg,
  user: UserSvg,
};

export const MenuIcon: React.FC<MenuIconProps> = ({ type }: MenuIconProps) => (
  <StyledMenuicon>{React.createElement(icons[type])}</StyledMenuicon>
);
