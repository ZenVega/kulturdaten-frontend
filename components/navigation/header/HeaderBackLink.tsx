import Link from 'next/link';
import styled from '@emotion/styled';

import { ChevronLeft } from 'react-feather';

const StyledA = styled.a`
  text-decoration: none;
  margin: 0.375rem 0 0.375rem 0.375rem;
  padding: 0.375rem 0.375rem 0.375rem 0;
  display: flex;
  align-items: center;
  color: inherit;
  font-size: var(--font-size-300);
  line-height: var(--line-height-300);
  font-weight: 700;
  border-radius: 0.75rem;
  background: transparent;

  &:hover {
    background: var(--grey-400);
  }

  span {
    font-size: var(--font-size-300);
    line-height: calc(var(--line-height-300) - 1px);
    border-bottom: 1px solid currentColor;
  }

  svg {
    display: inline-block;
    margin: 0 0.375rem 0 0;
    flex-shrink: 0;
    width: 1.5rem;
    height: 1.5rem;
  }
`;

export interface HeaderBackLinkProps {
  title: string;
  href: string;
}

export const HeaderBackLink: React.FC<HeaderBackLinkProps> = ({
  title,
  href,
}: HeaderBackLinkProps) => {
  return (
    <Link href={href} passHref>
      <StyledA title={title}>
        <ChevronLeft />
        <span>{title}</span>
      </StyledA>
    </Link>
  );
};
