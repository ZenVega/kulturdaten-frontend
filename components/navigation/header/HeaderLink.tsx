import Link from 'next/link';

import { useLocale, routes } from '../../../lib/routing';
import { useOrganizerId } from '../../../lib/useOrganizer';

export interface HeaderLinkProps {
  children: React.ReactElement<HTMLAnchorElement>;
}

export const HeaderLink: React.FC<HeaderLinkProps> = ({
  children,
}: {
  children: React.ReactElement;
}) => {
  const locale = useLocale();
  const organizerId = useOrganizerId();

  return (
    <Link href={routes.dashboard({ locale, query: { organizer: organizerId } })} passHref>
      {children}
    </Link>
  );
};
