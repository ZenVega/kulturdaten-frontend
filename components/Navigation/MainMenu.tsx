import styled from '@emotion/styled';
import React, { useMemo, useState } from 'react';

import { Header } from './Header';
import { MenuIcon } from './MenuIcon';
import { MenuLink, MenuLinkProps } from './MenuLink';

const StyledSubMenu = styled.div<{ active?: boolean }>`
  border: 1px solid var(--grey-400);
  border-top: none;
  width: 100%;
  background: ${({ active }) => (active ? 'var(--white)' : 'var(--grey-200)')};
  font-size: var(--font-size-300);
  line-height: var(--line-height-300);
  font-weight: 700;
`;

const StyledSubMenuHead = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0.375rem 0.75rem;
  border-bottom: 1px solid var(--grey-400);
`;

const StyledSubMenuTitle = styled.div``;
const StyledSubMenuIcon = styled.div``;

const StyledSubMenuContent = styled.div`
  padding: 1.125rem 0.75rem;
  display: grid;
  grid-row-gap: 0.75rem;
`;

const StyledSubMenuItem = styled.div``;

interface SubMenuProps {
  title: string;
  icon: React.ReactElement;
  links: React.ReactElement<MenuLinkProps>[];
  active?: boolean;
}

export const SubMenu: React.FC<SubMenuProps> = ({ title, icon, links, active }: SubMenuProps) => (
  <StyledSubMenu active={active}>
    <StyledSubMenuHead>
      <StyledSubMenuIcon>{icon}</StyledSubMenuIcon>
      <StyledSubMenuTitle>{title}</StyledSubMenuTitle>
    </StyledSubMenuHead>
    <StyledSubMenuContent>
      {links.map((item, index) => (
        <StyledSubMenuItem key={index}>{item}</StyledSubMenuItem>
      ))}
    </StyledSubMenuContent>
  </StyledSubMenu>
);

const StyledMainMenu = styled.div``;
const StyledMainMenuSubs = styled.div``;
const StyledMainMenuHeader = styled.div`
  position: sticky;
  top: 0;
  left: 0;
`;

interface MainMenuProps {
  subMenus: React.ReactElement<SubMenuProps>[];
  openState?: [boolean, React.Dispatch<React.SetStateAction<boolean>>];
  title: string;
  Link: React.FC<{ content: React.ReactElement }>;
}

export const MainMenu: React.FC<MainMenuProps> = ({
  subMenus,
  openState,
  title,
  Link,
}: MainMenuProps) => {
  const initialState = useState<boolean>(false);
  const [open, setOpen] = openState || initialState;

  return (
    <StyledMainMenu>
      <StyledMainMenuHeader>
        <Header title={title} Link={Link} openState={[open, setOpen]} />
      </StyledMainMenuHeader>
      <StyledMainMenuSubs>
        {subMenus.map((subMenu, index) => React.cloneElement(subMenu, { key: index }))}
      </StyledMainMenuSubs>
    </StyledMainMenu>
  );
};

export enum MenuIconName {
  start = 'start',
  organizer = 'organizer',
  offer = 'offer',
  location = 'location',
  user = 'user',
}

type MenuStructure = {
  title: string;
  icon: MenuIconName;
  links: {
    title: string;
    href: string;
    active?: boolean;
  }[];
}[];

export const useMainMenu = (
  structure: MenuStructure,
  title: string,
  Link: React.FC<{ content: React.ReactElement }>
): [React.ReactElement, [boolean, React.Dispatch<React.SetStateAction<boolean>>]] => {
  const [open, setOpen] = useState<boolean>(false);

  const activeSubMenu = useMemo(
    () =>
      structure.reduce<number>((activeIndex, item, index) => {
        for (let i = 0; i < item.links.length; i += 1) {
          if (item.links[i].active) {
            return index;
          }
        }
        return activeIndex;
      }, 0),
    [structure]
  );

  const subMenus = structure.map(({ title, icon, links }, index) => {
    const renderedLinks = links.map(({ title, href, active }, linkIndex) => (
      <MenuLink title={title} href={href} active={active} key={linkIndex} />
    ));
    return (
      <SubMenu
        active={activeSubMenu === index}
        title={title}
        icon={<MenuIcon type={icon} />}
        links={renderedLinks}
        key={index}
      />
    );
  });

  const renderedMainMenu = (
    <MainMenu openState={[open, setOpen]} subMenus={subMenus} title={title} Link={Link} />
  );

  return [renderedMainMenu, [open, setOpen]];
};
