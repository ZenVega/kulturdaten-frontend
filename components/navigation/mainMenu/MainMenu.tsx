import React, { useContext } from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/react';

import { Header } from '../header/Header';
import { HeaderLinkProps } from '../header/HeaderLink';
import { Sub, SubProps } from './Sub';
import { MenuIcon, MenuIconName } from './MenuIcon';
import { MenuLink, MenuLinkProps } from './MenuLink';
import { Breakpoint, useBreakpointOrWider, WindowContext } from '../../../lib/WindowService';
import { NavigationContext } from '../NavigationContext';
import { useKeyboard } from '../../../lib/useKeyboard';
import { LocaleSwitch } from '../LocaleSwitch';
import { Button, ButtonSize, ButtonVariant, IconPosition } from '../../button';
import { SubDivider } from './SubDivider';

const StyledMainMenu = styled.div<{ fullscreen?: boolean }>`
  background: var(--grey-200);
  display: flex;
  flex-direction: column;
  border: 1px solid var(--grey-400);
  border-bottom: none;
  position: relative;

  ${({ fullscreen }) =>
    fullscreen
      ? css`
          height: var(--app-height);
          overflow: hidden;
        `
      : ''};
`;

const StyledMainMenuContent = styled.div<{ show: boolean }>`
  display: none;
  visibility: hidden;
  overflow-y: auto;
  overflow-x: hidden;
  border-bottom: 1px solid var(--grey-400);
  flex-grow: 1;
  width: 100%;
  padding-bottom: env(safe-area-inset-bottom);

  ${({ show }) =>
    show
      ? css`
          display: block;
          visibility: inherit;
        `
      : ''}
`;

const StyledMainMenuSubs = styled.div``;
const StyledMainMenuHeader = styled.div`
  position: sticky;
  top: 0;
  left: 0;
`;

export interface MainMenuProps {
  subs: React.ReactElement<SubProps>[];
  title: string;
  Link: React.FC<HeaderLinkProps>;
}

export const MainMenu: React.FC<MainMenuProps> = ({ subs, title, Link }: MainMenuProps) => {
  const isWideOrWider = useBreakpointOrWider(Breakpoint.wide);
  const { mainMenuOpen, setMainMenuOpen } = useContext(NavigationContext);

  const { rendered } = useContext(WindowContext);

  useKeyboard(() => {
    setMainMenuOpen(false);
  }, ['Esc', 'Escape']);

  const showMenuContent = rendered && (isWideOrWider || mainMenuOpen);

  return (
    <StyledMainMenu fullscreen={showMenuContent}>
      <StyledMainMenuHeader>
        <Header title={title} Link={Link} />
      </StyledMainMenuHeader>
      <StyledMainMenuContent show={showMenuContent}>
        <StyledMainMenuSubs>
          {subs.map((sub, index) => React.cloneElement(sub, { key: index }))}
          <Sub items={[<LocaleSwitch key={1} />]} />
        </StyledMainMenuSubs>
      </StyledMainMenuContent>
    </StyledMainMenu>
  );
};

export enum MenuItem {
  link = 'link',
  button = 'button',
  divider = 'divider',
}

type MenuItemLink = {
  title: string;
  href: string;
  active?: boolean;
};

type MenuItemButton = {
  label: string;
  onClick: () => void;
  icon?: string;
  iconPosition?: IconPosition;
};

export type MenuStructure = {
  title?: string;
  icon?: MenuIconName;
  isActive?: boolean;
  items: {
    type: MenuItem;
    action?: MenuItemLink | MenuItemButton;
  }[];
}[];

export const useMainMenu = (
  structure: MenuStructure,
  title: string,
  Link: React.FC<HeaderLinkProps>
): React.ReactElement => {
  const { setMainMenuOpen } = useContext(NavigationContext);
  const subs = structure.map(({ title, icon, items, isActive }, index) => {
    const renderedItems = items?.map(({ type, action }, actionIndex) => {
      switch (type) {
        case MenuItem.link: {
          return <MenuLink key={actionIndex} {...(action as MenuLinkProps)} subMenuKey={index} />;
        }

        case MenuItem.button: {
          const { label, onClick, icon, iconPosition } = action as MenuItemButton;
          return (
            <Button
              onClick={() => {
                onClick();
                setMainMenuOpen(false);
              }}
              variant={ButtonVariant.minimal}
              size={ButtonSize.small}
              icon={icon}
              iconPosition={iconPosition}
            >
              {label}
            </Button>
          );
        }

        case MenuItem.divider: {
          return <SubDivider />;
        }

        default: {
          break;
        }
      }
    });

    return (
      <Sub
        title={title}
        icon={<MenuIcon type={icon} />}
        items={renderedItems}
        key={index}
        isActive={isActive}
      />
    );
  });

  const renderedMainMenu = <MainMenu subs={subs} title={title} Link={Link} />;

  return renderedMainMenu;
};

export const useMainMenuOverlayVisible = (): boolean => {
  const { mainMenuOpen } = useContext(NavigationContext);
  const isWideOrWider = useBreakpointOrWider(Breakpoint.wide);

  return mainMenuOpen && !isWideOrWider;
};
