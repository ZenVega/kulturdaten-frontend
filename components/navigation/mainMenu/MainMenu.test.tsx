import { render } from '../../../lib/test-utils';
import { MenuItem, useMainMenu } from './MainMenu';
import { MenuIconName } from './MenuIcon';
import { HeaderLink } from '../header/HeaderLink';
import React from 'react';

const testMenuStructure = [
  {
    title: 'Start',
    icon: MenuIconName.start,
    items: [
      {
        type: MenuItem.link,
        action: {
          title: 'Dashboard',
          href: '#',
          active: true,
        },
      },
      {
        type: MenuItem.link,
        action: {
          title: 'Benachrichtigungen',
          href: '#',
        },
      },
    ],
  },
  {
    title: 'Anbieter:innen',
    icon: MenuIconName.organizer,
    items: [
      {
        type: MenuItem.link,
        action: {
          title: 'Alle Anbieter:innen',
          href: '#',
        },
      },
      {
        type: MenuItem.link,
        action: {
          title: 'Meine Anbieter:innen',
          href: '#',
        },
      },
    ],
  },
  {
    title: 'Angebote',
    icon: MenuIconName.offer,
    items: [
      {
        type: MenuItem.link,
        action: {
          title: 'Alle Angebote',
          href: '#',
        },
      },
      {
        type: MenuItem.link,
        action: {
          title: 'Meine Angebote',
          href: '#',
        },
      },
    ],
  },
  {
    title: 'Orte',
    icon: MenuIconName.location,
    items: [
      {
        type: MenuItem.link,
        action: {
          title: 'Alle Orte',
          href: '#',
        },
      },
      {
        type: MenuItem.link,
        action: {
          title: 'Meine Orte',
          href: '#',
        },
      },
    ],
  },
  {
    title: 'Nutzer:in',
    icon: MenuIconName.user,
    items: [
      {
        type: MenuItem.link,
        action: {
          title: 'Mein Profil',
          href: '#',
        },
      },
      {
        type: MenuItem.link,
        action: {
          title: 'Meine Einstellungen',
          href: '#',
        },
      },
      {
        type: MenuItem.divider,
      },
      {
        type: MenuItem.button,
        action: {
          label: 'Abmelden',
          onClick: () => {
            //
          },
          icon: 'LogOut',
        },
      },
    ],
  },
  {
    items: [],
  },
];

const HookedMainMenu: React.FC = () => {
  const menu = useMainMenu(testMenuStructure, 'Test Title', HeaderLink);
  return menu;
};

test('MainMenu renders', () => {
  const { container } = render(
    <>
      <HookedMainMenu />
    </>
  );

  expect(container).toMatchSnapshot();
});
