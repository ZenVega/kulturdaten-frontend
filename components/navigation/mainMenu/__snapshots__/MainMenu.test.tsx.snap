// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu renders 1`] = `
<div>
  <div
    class="css-8eqs8y-StyledMainMenu ey6w9bd3"
  >
    <div
      class="css-1up9p1u-StyledMainMenuHeader ey6w9bd0"
    >
      <header
        class="css-1p8v3tm-HeaderContainer e1sn27871"
      >
        <button
          aria-label="Menü öffnen"
          class="css-1290inr-StyledMenuButton e91bmyz0"
          title="Menü öffnen"
        >
          <svg
            aria-labelledby="menu-button-icon"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="menu-button-icon"
            >
              Menü öffnen
            </title>
            <g>
              <path
                d="M5 8H19"
                stroke="#111111"
                stroke-linecap="round"
                stroke-width="2"
              />
            </g>
            <g>
              <path
                d="M19 16H5"
                stroke="#111111"
                stroke-linecap="round"
                stroke-width="2"
              />
            </g>
          </svg>
        </button>
        <a
          class="css-1kagrgy-StyledLink e1sn27870"
          href="/undefined"
        >
          Test Title
        </a>
      </header>
    </div>
    <div
      class="css-1q24dsu-StyledMainMenuContent ey6w9bd2"
    >
      <div
        class="css-7b8bp5-StyledMainMenuSubs ey6w9bd1"
      >
        <div
          class="css-1xq117s-StyledSub e136466r5"
        >
          <div
            class="css-15v88sk-StyledSubHead e136466r4"
          >
            <div
              class="css-1mfwjf3-StyledSubIcon e136466r2"
            >
              <div
                class="css-1qgbttw-StyledMenuicon e1g2h9ar0"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.25 9.75L12 4.5L18.75 9.75V18C18.75 18.3978 18.592 18.7794 18.3107 19.0607C18.0294 19.342 17.6478 19.5 17.25 19.5H6.75C6.35218 19.5 5.97064 19.342 5.68934 19.0607C5.40804 18.7794 5.25 18.3978 5.25 18V9.75Z"
                    stroke="#111111"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                  <path
                    d="M9.75 19.5V12H14.25V19.5"
                    stroke="#111111"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
            <div
              class="css-1brgdsk-StyledSubTitle e136466r3"
            >
              Start
            </div>
          </div>
          <div
            class="css-1h9g97m-StyledSubContent e136466r1"
          >
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-1wahcwk-StyledA e1l78fsc0"
                href="/#"
                title="Dashboard"
              >
                <svg
                  fill="none"
                  height="18"
                  viewBox="0 0 18 18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.75 9H14.25"
                    stroke="#111111"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                  <path
                    d="M9 3.75L14.25 9L9 14.25"
                    stroke="#111111"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                Dashboard
              </a>
            </div>
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-cjivcy-StyledA e1l78fsc0"
                href="/#"
                title="Benachrichtigungen"
              >
                
                Benachrichtigungen
              </a>
            </div>
          </div>
        </div>
        <div
          class="css-nvofqz-StyledSub e136466r5"
        >
          <div
            class="css-15v88sk-StyledSubHead e136466r4"
          >
            <div
              class="css-1mfwjf3-StyledSubIcon e136466r2"
            >
              <div
                class="css-1qgbttw-StyledMenuicon e1g2h9ar0"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17.1823 4.32248C16.6842 2.46356 14.3606 1.84094 12.9997 3.20177L3.20178 12.9997C1.84095 14.3606 2.46357 16.6842 4.3225 17.1823L17.7068 20.7686C19.5657 21.2667 21.2667 19.5657 20.7686 17.7067L17.1823 4.32248Z"
                    fill="#E60032"
                    stroke="url(#paint2_linear)"
                  />
                  <defs>
                    <lineargradient
                      gradientUnits="userSpaceOnUse"
                      id="paint2_linear"
                      x1="8.19702"
                      x2="19.697"
                      y1="8.6969"
                      y2="20.1969"
                    >
                      <stop
                        stop-color="#E58EA1"
                      />
                      <stop
                        offset="1"
                        stop-color="#9F0B2B"
                      />
                    </lineargradient>
                  </defs>
                </svg>
              </div>
            </div>
            <div
              class="css-1brgdsk-StyledSubTitle e136466r3"
            >
              Anbieter:innen
            </div>
          </div>
          <div
            class="css-1h9g97m-StyledSubContent e136466r1"
          >
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-cjivcy-StyledA e1l78fsc0"
                href="/#"
                title="Alle Anbieter:innen"
              >
                
                Alle Anbieter:innen
              </a>
            </div>
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-cjivcy-StyledA e1l78fsc0"
                href="/#"
                title="Meine Anbieter:innen"
              >
                
                Meine Anbieter:innen
              </a>
            </div>
          </div>
        </div>
        <div
          class="css-nvofqz-StyledSub e136466r5"
        >
          <div
            class="css-15v88sk-StyledSubHead e136466r4"
          >
            <div
              class="css-1mfwjf3-StyledSubIcon e136466r2"
            >
              <div
                class="css-1qgbttw-StyledMenuicon e1g2h9ar0"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#273B8A"
                    height="17"
                    rx="2.5"
                    stroke="url(#paint1_linear)"
                    transform="rotate(-15 1.58978 5.98959)"
                    width="17"
                    x="1.58978"
                    y="5.98959"
                  />
                  <defs>
                    <lineargradient
                      gradientUnits="userSpaceOnUse"
                      id="paint1_linear"
                      x1="2.20215"
                      x2="18.2021"
                      y1="6.34314"
                      y2="22.3431"
                    >
                      <stop
                        stop-color="#6984EF"
                      />
                      <stop
                        offset="1"
                        stop-color="#000B37"
                      />
                    </lineargradient>
                  </defs>
                </svg>
              </div>
            </div>
            <div
              class="css-1brgdsk-StyledSubTitle e136466r3"
            >
              Angebote
            </div>
          </div>
          <div
            class="css-1h9g97m-StyledSubContent e136466r1"
          >
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-cjivcy-StyledA e1l78fsc0"
                href="/#"
                title="Alle Angebote"
              >
                
                Alle Angebote
              </a>
            </div>
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-cjivcy-StyledA e1l78fsc0"
                href="/#"
                title="Meine Angebote"
              >
                
                Meine Angebote
              </a>
            </div>
          </div>
        </div>
        <div
          class="css-nvofqz-StyledSub e136466r5"
        >
          <div
            class="css-15v88sk-StyledSubHead e136466r4"
          >
            <div
              class="css-1mfwjf3-StyledSubIcon e136466r2"
            >
              <div
                class="css-1qgbttw-StyledMenuicon e1g2h9ar0"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    fill="#46948C"
                    r="9.5"
                    stroke="url(#paint0_linear)"
                  />
                  <defs>
                    <lineargradient
                      gradientUnits="userSpaceOnUse"
                      id="paint0_linear"
                      x1="4.5"
                      x2="19"
                      y1="5"
                      y2="19.5"
                    >
                      <stop
                        stop-color="#6FC1B9"
                      />
                      <stop
                        offset="1"
                        stop-color="#1B645C"
                      />
                    </lineargradient>
                  </defs>
                </svg>
              </div>
            </div>
            <div
              class="css-1brgdsk-StyledSubTitle e136466r3"
            >
              Orte
            </div>
          </div>
          <div
            class="css-1h9g97m-StyledSubContent e136466r1"
          >
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-cjivcy-StyledA e1l78fsc0"
                href="/#"
                title="Alle Orte"
              >
                
                Alle Orte
              </a>
            </div>
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-cjivcy-StyledA e1l78fsc0"
                href="/#"
                title="Meine Orte"
              >
                
                Meine Orte
              </a>
            </div>
          </div>
        </div>
        <div
          class="css-nvofqz-StyledSub e136466r5"
        >
          <div
            class="css-15v88sk-StyledSubHead e136466r4"
          >
            <div
              class="css-1mfwjf3-StyledSubIcon e136466r2"
            >
              <div
                class="css-1qgbttw-StyledMenuicon e1g2h9ar0"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 18.75V17.25C18 16.4544 17.6839 15.6913 17.1213 15.1287C16.5587 14.5661 15.7956 14.25 15 14.25H9C8.20435 14.25 7.44129 14.5661 6.87868 15.1287C6.31607 15.6913 6 16.4544 6 17.25V18.75"
                    stroke="#111111"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                  <path
                    d="M12 11.25C13.6569 11.25 15 9.90685 15 8.25C15 6.59315 13.6569 5.25 12 5.25C10.3431 5.25 9 6.59315 9 8.25C9 9.90685 10.3431 11.25 12 11.25Z"
                    stroke="#111111"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
            <div
              class="css-1brgdsk-StyledSubTitle e136466r3"
            >
              Nutzer:in
            </div>
          </div>
          <div
            class="css-1h9g97m-StyledSubContent e136466r1"
          >
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-cjivcy-StyledA e1l78fsc0"
                href="/#"
                title="Mein Profil"
              >
                
                Mein Profil
              </a>
            </div>
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <a
                class="css-cjivcy-StyledA e1l78fsc0"
                href="/#"
                title="Meine Einstellungen"
              >
                
                Meine Einstellungen
              </a>
            </div>
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <div
                class="css-hgfigs-SubDivider e1pe1px00"
              />
            </div>
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <button
                class="css-sryp67-StyledButton-minimal e1asxjmr2"
                color="default"
              >
                <span
                  class="css-1ts8fox-StyledButtonSpan e1asxjmr1"
                >
                  Abmelden
                </span>
                <div
                  class="css-1wucyrk-StyledButtonIcon e1asxjmr0"
                >
                  <svg
                    fill="none"
                    height="16"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
                    />
                    <polyline
                      points="16 17 21 12 16 7"
                    />
                    <line
                      x1="21"
                      x2="9"
                      y1="12"
                      y2="12"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="css-nvofqz-StyledSub e136466r5"
        >
          <div
            class="css-1h9g97m-StyledSubContent e136466r1"
          />
        </div>
        <div
          class="css-nvofqz-StyledSub e136466r5"
        >
          <div
            class="css-1h9g97m-StyledSubContent e136466r1"
          >
            <div
              class="css-w03ot7-StyledSubItem e136466r0"
            >
              <div
                class="css-1dc2ar3-StyledSelectContainer e1vif2fk4"
              >
                <label
                  class="css-1lmhznq-StyledLabel e1d1vdpj0"
                  for="locale"
                >
                  Sprache / Language
                </label>
                <div
                  class="css-15csdm1-StyledSelectAndChevron e1vif2fk0"
                >
                  <select
                    aria-label="Sprache wählen, choose Language"
                    class="css-1jajov3-StyledSelect-minimal e1vif2fk3"
                    id="locale"
                  >
                    <option
                      value="de-DE"
                    >
                      Deutsch
                    </option>
                    <option
                      value="en-DE"
                    >
                      English
                    </option>
                  </select>
                  <div
                    class="css-o4rktt-StyledSelectIcon e1vif2fk2"
                  >
                    <svg
                      fill="none"
                      height="18"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                      />
                      <line
                        x1="2"
                        x2="22"
                        y1="12"
                        y2="12"
                      />
                      <path
                        d="M12 2a15.3 15.3 0 0 1 4 10 15.3 15.3 0 0 1-4 10 15.3 15.3 0 0 1-4-10 15.3 15.3 0 0 1 4-10z"
                      />
                    </svg>
                  </div>
                  <div
                    class="css-1hj2sr-StyledSelectIcon-StyledSelectChevron e1vif2fk1"
                  >
                    <svg
                      fill="none"
                      height="18"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <polyline
                        points="6 9 12 15 18 9"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;
